src/psyclone/psyir/transformations/reference2arrayrange_trans.py

```diff
@@ -138,7 +165,23 @@ def apply(self, node, options=None, **kwargs):
         '''
         self.validate(node, **kwargs)
 
+        # We have validated that it is a reference to a symbol with a datatype
+        # that is not UnresolvedType
         symbol = node.symbol
+
+        # The following cases do not need expansions
+        if type(node) is ArrayReference:
+            return
+        if type(node) is ArrayOfStructuresReference:
+            return
+        if node.parent and isinstance(node.parent, Call):
+            if node.position == 0:
+                return
```

Comment:
Comment please. Perhaps `if node is node.parent.routine` would make this clearer (and in the other case where I asked for a comment)?